backend/src/main/java/com/redhat/cloud/notifications/routers/sources/SourcesXRHIDHeaderFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.redhat.cloud.notifications.routers.sources;
+
+import com.redhat.cloud.notifications.Constants;
+import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * This class will grab the incoming "x-rh-identity" header and forward it along when sending requests to Sources. We
+ * could have used the default header factory and then the "propagateHeaders" property, but in order to avoid any
+ * issues with future header factories that might want to do that and not forward the "x-rh-identity" header, this
+ * simple class is built instead.
+ */
+public class SourcesXRHIDHeaderFactory implements ClientHeadersFactory {
+
+    /**
+     * Grabs the incoming "x-rh-identity" header and returns it.
+     * @param incomingHeaders the incoming headers. They are left untouched.
+     * @param outgoingHeaders the outgoing headers. They are left untouched.
+     * @return a map containing the "x-rh-identity" header.
+     */
+    @Override
+    public MultivaluedMap<String, String> update(final MultivaluedMap<String, String> incomingHeaders, final MultivaluedMap<String, String> outgoingHeaders) {
+        final var headers = new MultivaluedHashMap<String, String>();
+        headers.add(Constants.X_RH_IDENTITY_HEADER, incomingHeaders.getFirst(Constants.X_RH_IDENTITY_HEADER));
+
+        return headers;
+    }
```

Comment:
The idea here was to simply forward the `x-rh-identity` header, since that's the only one that Sources requires for the request to be authenticated. By adding all the `outgoingHeaders`, wouldn't we be sending also headers that might be specific for Notifications too? :thinking:

Reply:
`outgoingHeaders` is currently empty because the `SourcesService` methods are not annotated with `@ClientHeaderParam` or `@HeaderParam`. This will probably never change and your code will most likely work in the future 👍

However, if this class is copied (which happens a lot) to build something else, then the copied class may remove the outgoing headers unintentionally and the dev may waste time understanding why this is happening. So I would merge the outgoing headers here (it won't hurt) even if it's not needed for now 😃

Reply:
Ah, I see. Totally understandable. I included the suggestion too! :+1: